### PR TITLE
Do not pause scheduling on waiting allocations

### PR DESCRIPTION
**Pull request by Krock21**
Do not pause scheduling on waiting allocations

Please cherry-pick to stable/24.1

### CODE REVIEW

File: yt/yt/server/scheduler/node_shard.cpp [hunk 1]
```diff
@@ -597,17 +597,10 @@ void TNodeShard::DoProcessHeartbeat(const TScheduler::TCtxNodeHeartbeatPtr& cont
     }
 
     bool skipScheduleAllocations = false;
-    if (hasWaitingAllocations || isThrottlingActive) {
```
Comment (k-pogorelov) [inline yt/yt/server/scheduler/node_shard.cpp:600]: Let's remove hasWaitingAllocations completely

**Review by k-pogorelov — CHANGES_REQUESTED** [review]

File: yt/yt/server/scheduler/node_shard.cpp (hunk 1 above)
Comment (Krock21) [reply yt/yt/server/scheduler/node_shard.cpp:600]: We need it to pass it by reference to `ProcessHeartbeatAllocations` on line 596

Comment (k-pogorelov) [reply yt/yt/server/scheduler/node_shard.cpp:600]: It is true, but look, what we pass it in ProcessHeartbeatAllocations for :)

**Review by k-pogorelov — APPROVED** [review]

File: yt/yt/server/scheduler/node_shard.cpp (hunk 1 above)
Comment (Krock21) [reply yt/yt/server/scheduler/node_shard.cpp:600]: Removed it from ProcessHeartbeatAllocations too

**Pull request closed without merging**